buildenv/jenkins/ARTIFACTORY.md

```diff
@@ -0,0 +1,200 @@
+    proxy_set_header    X-Artifactory-Override-Base-Url $http_x_forwarded_proto://$host:$server_port/artifactory;
+    proxy_set_header    X-Forwarded-Port  $server_port;
+    proxy_set_header    X-Forwarded-Proto $http_x_forwarded_proto;
+    proxy_set_header    Host              $http_host;
+    proxy_set_header    X-Forwarded-For   $proxy_add_x_forwarded_for;
+    }
+
+    listen [::]:443 ssl ipv6only=on; # managed by Certbot
+    listen 443 ssl; # managed by Certbot
+    ssl_certificate /etc/letsencrypt/live/140-211-168-230-openstack.osuosl.org/fullchain.pem; # managed by Certbot
+    ssl_certificate_key /etc/letsencrypt/live/140-211-168-230-openstack.osuosl.org/privkey.pem; # managed by Certbot
+    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
+    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
+
+}
+
+server {
+    if ($host = 140-211-168-230-openstack.osuosl.org) {
+        return 301 https://$host$request_uri;
+    } # managed by Certbot
+
+
+    listen 80 ;
+    listen [::]:80 ;
+
+    server_name 140-211-168-230-openstack.osuosl.org;
+    return 404; # managed by Certbot
+}
+```
+All of the lines with the `# managed by Certbot` were added when adding in the ssl certificate. 
+
+Those lines do not need to be added as they were automatically added by certbot
+
+I am getting my certificate from Let’s Encrypt and their install tool certbot. Firstly, I installed certbot:
+```
+sudo apt-get update
+sudo apt-get install software-properties-common
+sudo add-apt-repository universe
+sudo add-apt-repository ppa:certbot/certbot
+sudo apt-get update
+sudo apt-get install certbot python-certbot-nginx
+```
+Then I used the certbot automated script to add the certificates and to automatically renew the certificates: 
+```
+sudo certbot –nginx
+```
+Follow the onscreen instructions and everything should work out. 
+
+# Open the Ports
+Log into OpenStack and go to network security groups
+Click on `http` (if it is not created, create the rule)
+Create a new rule with these settings
+```
+Custome TCP Rule
+Ingres
+Port
+443
+CIDR
+0.0.0.0/0
+```
+Create another rule for port 80 if you enabled port forward in the certbot step
+
+Then go to compute -> instances and click on ` eclipse-openj9-proxy`
+
+Edit instance -> security group
+Add http to the instance security groups
+
+
+
+# Configure Artifactory 
+To start the Artifactory server, run the command:
+```
+/home/jenkins/artifactory/artifactory-oss-6.5.2/bin/artifactoryctl start
+```
+
+On your local browser, go to `http://140-211-168-230-openstack.osuosl.org/` and follow the onscreen instructions.
+The admin password is in the secrets repo and the repository that is used is ‘generic’ with the name being “ci-eclipse-openj9”. 
+
+Next, click on the `Welcome, admin` tab and select `Add User`
+Insert the username `Jenkins` and the email address `j9build@ca.ibm.com`. Insert the password as indicated in the secrets file.
```

Comment:
Is the email ever needed/used? We may want to use the genie-openj9 email?

Reply:
I do not believe that the email will be used.